src/main/java/openperipheral/addons/sensors/AdapterSensor.java

```diff
@@ -191,16 +272,73 @@ public IMetaProviderProxy getPlayerByUUID(ISensorEnvironment env,
 					else if (block.getMaterial().isSolid()) type = "SOLID";
 					else type = "UNKNOWN";
 
+					final String color;
+					if (block == null) {
+						color = "f"; // Default black -- same as air.
+					} else {
+						int rgb = block.getMapColor(block.getDamageValue(world, bx, by, bz)).colorValue;
+						color = ComputerColors.getClosestColor(rgb).code;
+					}
+
 					Map<String, Object> tmp = Maps.newHashMap();
 					tmp.put("x", x);
 					tmp.put("y", y);
 					tmp.put("z", z);
 					tmp.put("type", type);
+					tmp.put("color", color);
 					results.put(++i, tmp);
 
 				}
 			}
 		}
 		return results;
 	}
+
+	@ScriptCallable(returnTypes = ReturnType.TABLE, description = "Get a table of information about a single block in the surrounding area. Includes map color and whether each block is UNKNOWN, AIR, LIQUID, or SOLID.")
+	public Map<String, Object> sonicScanTarget(ISensorEnvironment env,
+			@Arg(name = "xOffset", description = "The target's offset from the sensor on the X-Axis.") int xOff,
+			@Arg(name = "yOffset", description = "The target's offset from the sensor on the Y-Axis.") int yOff,
+			@Arg(name = "zOffset", description = "The target's offset from the sensor on the Z-Axis.") int zOff) {
+
+		int range = 1 + env.getSensorRange() / 2;
+		World world = env.getWorld();
+		Vec3 sensorPos = env.getLocation();
+		int sx = MathHelper.floor_double(sensorPos.xCoord);
+		int sy = MathHelper.floor_double(sensorPos.yCoord);
+		int sz = MathHelper.floor_double(sensorPos.zCoord);
+
+		final int rangeSq = range * range;
+
+		final int bx = sx + xOff;
+		final int by = sy + yOff;
+		final int bz = sz + zOff;
+		if (!world.blockExists(bx, by, bz)) return null;
+
+		final int distSq = xOff * xOff + yOff * yOff + zOff * zOff;
+		if (distSq == 0 || distSq > rangeSq) return null;
+		Block block = world.getBlock(bx, by, bz);
+
+		String type;
+		if (block == null || world.isAirBlock(bx, by, bz)) type = "AIR";
+		else if (block.getMaterial().isLiquid()) type = "LIQUID";
+		else if (block.getMaterial().isSolid()) type = "SOLID";
+		else type = "UNKNOWN";
+
+		final String color;
+		if (block == null) {
+			color = "f"; // Default black -- same as air.
+		} else {
+			int rgb = block.getMapColor(block.getDamageValue(world, bx, by, bz)).colorValue;
+			color = ComputerColors.getClosestColor(rgb).code;
+		}
+
+		Map<String, Object> tmp = Maps.newHashMap();
+		tmp.put("x", xOff);
+		tmp.put("y", yOff);
+		tmp.put("z", zOff);
+		tmp.put("type", type);
+		tmp.put("color", color);
+
+		return tmp;
+	}
```

Comment:
Parts of if look like copy of `sonicScan`. Please extract common parts to function.